src/WebJobs.Extensions.DurableTask/Listener/DurableTaskTargetScaler.cs

```diff
@@ -38,46 +39,58 @@ public DurableTaskTargetScaler(string functionId, DurableTaskMetricsProvider met
 
         public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
         {
-            // This method is only invoked by the ScaleController, so it doesn't run in the Functions Host process.
-            var metrics = await this.metricsProvider.GetMetricsAsync();
-
-            // compute activityWorkers: the number of workers we need to process all activity messages
-            var workItemQueueLength = metrics.WorkItemQueueLength;
-            double activityWorkers = Math.Ceiling(workItemQueueLength / (double)this.MaxConcurrentActivities);
-
-            var serializedControlQueueLengths = metrics.ControlQueueLengths;
-            var controlQueueLengths = JsonConvert.DeserializeObject<IReadOnlyList<int>>(serializedControlQueueLengths);
-
-            var controlQueueMessages = controlQueueLengths.Sum();
-            var activeControlQueues = controlQueueLengths.Count(x => x > 0);
-
-            // compute orchestratorWorkers: the number of workers we need to process all orchestrator messages.
-            // We bound this result to be no larger than the partition count
-            var upperBoundControlWorkers = Math.Ceiling(controlQueueMessages / (double)this.MaxConcurrentOrchestrators);
-            var orchestratorWorkers = Math.Min(activeControlQueues, upperBoundControlWorkers);
-
-            int numWorkersToRequest = (int)Math.Max(activityWorkers, orchestratorWorkers);
-            this.scaleResult.TargetWorkerCount = numWorkersToRequest;
-
-            // When running on ScaleController V3, ILogger logs are forwarded to the ScaleController's Kusto table.
-            // This works because this code does not execute in the Functions Host process, but in the ScaleController process,
-            // and the ScaleController is injecting it's own custom ILogger implementation that forwards logs to Kusto.
-            var scaleControllerLog = $"Target worker count for {this.functionId}: {numWorkersToRequest}. " +
-                $"Metrics used: workItemQueueLength={workItemQueueLength}. controlQueueLengths={serializedControlQueueLengths}. " +
-                $"maxConcurrentOrchestrators={this.MaxConcurrentOrchestrators}. maxConcurrentActivities={this.MaxConcurrentActivities}";
-
-            // target worker count should never be negative
-            if (numWorkersToRequest < 0)
+            DurableTaskTriggerMetrics? metrics = null;
+            try
             {
```

Comment:
To fall back to regular scale for Netherite and MSSQL you need to throw `NonSupportedException`:

[azure-sdk-for-net/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusTargetScaler.cs at main · Azure/azure-sdk-for-net (github.com)](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusTargetScaler.cs#L61)

 
Do you have this implemented for durable?